app/src/main/java/dev/patrickgold/florisboard/ime/text/TextInputManager.kt

```diff
@@ -667,6 +673,65 @@ class TextInputManager private constructor() : CoroutineScope by MainScope(), In
         }
     }
 
+    /**
+     * Handles a [KeyCode.KANA_SWITCHER] event
+     */
+    private fun handleKanaSwitch() {
+        activeState.isKanaKata = !activeState.isKanaKata
+        florisboard.dispatchCurrentStateToInputUi()
+    }
```

Comment:
What happened to the kana switcher? It does not cycle through the modes anymore. Based on commit 0ac30ce746135a8fedcec2c84f25eaccd95717cc I can see you wanted to remove half_kata from the cycle. The problem is, that you don't set the char width back to full width, so now there are two possible cycles possible:
- Hira -> Kata (What you want)
- Hira -> Half kata (happens if I was previously in the symbols layout and the char width is still on half width.

Should this be unintended, I suggest this change:

```suggestion
    private fun handleKanaSwitch() {
        activeState.isKanaKata = !activeState.isKanaKata
        activeState.isCharHalfWidth = false
        florisboard.dispatchCurrentStateToInputUi()
    }
```

Reply:
Yeah, i didn't catch that. Previously that function delegates to the other kana type functions which dealt with the character width